### PR TITLE
Improve the fetching of the contact's baseurl

### PR DESCRIPTION
**Pull request by annando**
We now check the existence of the baseurl field when the contact is inserted. Also we store the baseurl for some system actors that are identical to the baseurl.

### CODE REVIEW

File: src/Model/Contact.php [hunk 1]
```diff
@@ -1372,6 +1377,7 @@ public static function getIdForURL(string $url = null, int $uid = 0, $update = n
 			$fields = [
 				'uid'       => $uid,
 				'url'       => $data['url'],
+				'baseurl'   => $data['baseurl'] ?? '',
```
Comment (MrPetovan) [inline src/Model/Contact.php:1380]: What about the related `gsid` field? The `baseurl` is used to show the Ignore Server label but it's the `gsid` we need to actually perform the action.

File: src/Model/Contact.php [hunk 2]
```diff
@@ -3252,6 +3258,7 @@ public static function addRelationship(array $importer, array $contact, array $d
 				'created'  => DateTimeFormat::utcNow(),
 				'url'      => $url,
 				'nurl'     => Strings::normaliseLink($url),
+				'baseurl'  => $pub_contact['baseurl'] ?? '',
```
Comment (MrPetovan) [inline src/Model/Contact.php:3261]: Same question about the related `gsid` field.

File: src/Model/Contact.php (hunk 1 above)
Comment (annando) [reply src/Model/Contact.php:1380]: In `Contact::insert` we are creating the `gsid` based on the `baseurl`.

File: src/Model/Contact.php (hunk 2 above)
Comment (annando) [reply src/Model/Contact.php:3261]: We only need the `baseurl` to fetch the `gsid`.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**